### ISSUE TITLE
Middle-mouse-clicking a wrangled tab line to re-open that tab and allowing to re-open another? (without closing the pop-up)

### ISSUE THREAD

**Issue by Wizek** [jacobSingh/tabwrangler#144]
Is it clear what I mean from the title, or would you prefer more details?

In case it's clear, I'd be open to implementing the feature myself and submitting a pull request too.
Would you be open to a PR for such a feature?
Any related thoughts?

**Comment by ssorallen** [comment]
I believe that's how it works right now when you click the text link in the tab row. Is it behaving differently than you'd like?

**Comment by Wizek** [comment]
Yes, differently. I'd like if the pop-up stayed open, and currently it closes after the middle click.

**Comment by ssorallen** [comment]
It should currently keep the popup open if you left click. Does left clicking close the popup for you?

**Comment by Wizek** [comment]
Oh. How true. It stays open upon just left-clicking. For some reason I never thought to try (or maybe just not recently?), and expected that if anything, middle click will keep it open.

Wouldn't it make sense to swap the behaviour of left clicking and middle clicking? That would be more inline with how browsers in general handle clicking on links, wouldn't it?
 
> If I left-click a link, I get to visit it immediately, navigating away from my current page.
> If I middle click a link then it opens in the background tab while also letting me keep the current tab, potentially allowing me to middle-click other links too in succession.

**Comment by ssorallen** [comment]
> Wouldn't it make sense to swap the behaviour of left clicking and middle clicking?

That seems reasonable to me. Users might be used to the behavior at this point though, and I don't want clicking to suddenly feel broken. We could make it an option and keep the behavior the same for current users but default to your suggestion for new users.

**Comment by Wizek** [comment]
> We could make it an option and keep the behavior the same for current users but default to your suggestion for new users.

I like that idea!

**Comment by joeytwiddle** [comment]
A counter argument: Left-click should do whatever is **the most common use case** for users. That is the friendliest way for users to start using the product.

So if most users open _more than one_ link when they use the Tab Wrangler popup, then I think keeping the popup open should be the default left-click behaviour.

To know whether that is true or not, we would need to start collecting user data! (With a suitable opt-in agreement of course.)

Having said that, there certainly is merit in your argument to keep the behaviour similar to how links work elsewhere in the browser.

----

Personally I had never thought of trying middle-click. I [suspect](control click or middle click site:ux.stackexchange.com) most users are more likely to try Ctrl-click as an alternative (or Cmd-click on Mac). But I could be wrong about that.

If you want to make the different behaviours more discoverable, you could show a popup while a link is being hovered: "Open page in new tab | Middle-click: Open page in new tab and close Tab Wrangler | Ctrl-click: Open page in current tab and close Tab Wrangler" etc.

**Comment by KTamas** [comment]
if someone implements it among the lines of what @joeytwiddle describes, i'm open to it, but this is not a priority.
